### PR TITLE
Gbou/fullscreen table component tests PROD-10289

### PR DESCRIPTION
**Pull request by Guismus**
Before creating a PR, please check that:
- [ ] Code is clean
- [ ] Tests are still working (cypress tests and `yarn test`)
- [ ] Changes don't cause new errors or warnings in browser console
- [ ] Documentation is up-to-date
- [ ] Breaking changes are clearly identified with [conventional commits](https://kapeli.com/cheat_sheets/Conventional_Commits.docset/Contents/Resources/Documents/index)

### CODE REVIEW

**Review by ardvi — COMMENTED** [review]
one comment but LGTM :+1:

File: cypress/commons/actions/generic/TableParameters.js [hunk 1]
```diff
@@ -79,6 +83,10 @@ function getExportConfirmButton() {
   return getExportDialog().find(GENERIC_SELECTORS.genericComponents.table.export.confirmButton);
 }
 
+function getFullscreenButton(tableParameterElement) {
+  return getGrid(tableParameterElement).find(GENERIC_SELECTORS.genericComponents.table.toolbar.fullscreen);
```
Comment (csm-thu) [inline cypress/commons/actions/generic/TableParameters.js:87]: Could we rename the last key from `fullscreen` to `fullscreenButton`?

File: cypress/commons/actions/generic/TableParameters.js [hunk 2]
```diff
@@ -142,6 +150,10 @@ function setFileExportName(fileName) {
   getExportFileNameInput().type('{selectAll}{backspace}' + fileName + '{enter}');
 }
 
+function triggerFullscreenButton(tableParameterElement) {
```
Comment (csm-thu) [inline cypress/commons/actions/generic/TableParameters.js:153]: Could we rename this function to `toggleFullscreen`?

File: cypress/commons/actions/generic/TableParameters.js [hunk 3]
```diff
@@ -164,7 +176,14 @@ function clearStringCell(getTableElement, colName, rowIndex) {
   return getCell(getTableElement(), colName, rowIndex);
 }
 
+function scrollTable(getTableElement, position) {
```
Comment (csm-thu) [inline cypress/commons/actions/generic/TableParameters.js:179]: Could we rename this function to `scrollTo` or `scrollToPosition`?

Also, could you add a comment to explain what value is expected for the `position` argument? When reading this code, I have absolutely no idea if this should be a position in pixels, an integer for a row number, a DOM element selected by cypress, or something else... :sweat_smile:

File: cypress/integration/brewery/TableParameters-fullscreen.spec.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    BreweryParameters.getCustomersFullscreenButton().should('not.exist');
+    BreweryParameters.importCustomersTableData('customers.csv');
+    TableParameters.importFile(getVipCustomersTable(), 'customers.csv');
+
+    BreweryParameters.getCustomersFullscreenButton().should('be.visible');
+    BreweryParameters.enterCustomersFullscreenButton();
+    TableParameters.editStringCell(getFullscreenTable, 'age', 0, '11', true).should('have.text', '11');
+    TableParameters.editStringCell(getFullscreenTable, 'canDrinkAlcohol', 0, 'true', true).should('have.text', 'true');
+    TableParameters.editStringCell(getFullscreenTable, 'favoriteDrink', 0, 'Wine', true).should('have.text', 'Wine');
+    TableParameters.editStringCell(getFullscreenTable, 'birthday', 0, '02/05/2012', true).should(
```
Comment (csm-thu) [inline cypress/integration/brewery/TableParameters-fullscreen.spec.js:51]: You're defining 5 parameters when calling `editStringCell`, but it has only 4:
`function editStringCell(getTableElement, colName, rowIndex, newValue)`

If it is not necessary, could you remove the last argument `true`?

File: cypress/integration/brewery/TableParameters-fullscreen.spec.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  it('Edit a cell, and check if value as change in and out of fullscreen', () => {
+    Scenarios.getScenarioViewTab(60).should('be.visible');
+    ScenarioParameters.expandParametersAccordion();
+    BreweryParameters.switchToCustomersTab();
+    BreweryParameters.getCustomersTable().should('be.visible');
+    BreweryParameters.getCustomersTableLabel().should('be.visible').should('have.text', 'Customers');
+    BreweryParameters.getCustomersTableGrid().should('exist');
+    BreweryParameters.getCustomersFullscreenButton().should('not.exist');
+    BreweryParameters.importCustomersTableData('customers.csv');
+    TableParameters.importFile(getVipCustomersTable(), 'customers.csv');
```
Comment (csm-thu) [inline cypress/integration/brewery/TableParameters-fullscreen.spec.js:44]: Why do we import the same file _customers.csv_ for the "vip customers" table? Its columns are different from the ones in the customers table, right?

File: cypress/integration/brewery/TableParameters-fullscreen.spec.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+      '02/05/2012'
+    );
+    BreweryParameters.exitCustomersFullscreenButton();
+
+    BreweryParameters.getCustomersTableCell('age', 0).should('have.text', '11');
+    BreweryParameters.getCustomersTableCell('canDrinkAlcohol', 0).should('have.text', 'true');
+    BreweryParameters.getCustomersTableCell('favoriteDrink', 0).should('have.text', 'Wine');
+    BreweryParameters.getCustomersTableCell('birthday', 0).should('have.text', '02/05/2012');
+
+    TableParameters.getCell(getVipCustomersTable(), 'vipLevel', 0).should('have.text', '10');
```
Comment (csm-thu) [inline cypress/integration/brewery/TableParameters-fullscreen.spec.js:62]: I can't find where this value "10" comes from, but maybe your answer to my previous comment will answer this question too :slightly_smiling_face:

**Review by csm-thu — CHANGES_REQUESTED** [review]
Some remarks and questions, and some changes suggested :slightly_smiling_face:

File: cypress/integration/brewery/TableParameters-fullscreen.spec.js [hunk 7]
```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import { Scenarios, ScenarioParameters, TableParameters, Login } from '../../commons/actions';
+import { BreweryParameters } from '../../commons/actions/brewery';
+import { stub } from '../../commons/services/stubbing';
+import { SOLUTIONS } from '../../fixtures/stubbing/TableParameters-fullscreen/solutions';
+
+const getFullscreenTable = TableParameters.getFullscreenTable;
```
[the remark this answers is not in the document]
Comment (Guismus) [reply cypress/integration/brewery/TableParameters-fullscreen.spec.js:9]: getFullscreenTable is a function which do not need argument, and he is only used has parameter argument for some function. Reduce the size of this parameter allow to bypass some to much column problem on some line and so be more lisible

File: cypress/commons/actions/brewery/BreweryParameters.js [hunk 8]
```diff
@@ -175,6 +175,15 @@ function getCustomersImportButton() {
...
   return TableParameters.getExportButton(getCustomersTable());
 }
+function getCustomersFullscreenButton() {
+  return TableParameters.getFullscreenButton(getCustomersTable());
+}
+function enterCustomersFullscreenButton() {
+  return TableParameters.triggerFullscreenButton(getCustomersTable());
+}
+function exitCustomersFullscreenButton() {
+  return TableParameters.triggerFullscreenButton(TableParameters.getFullscreenTable());
```
[the remark this answers is not in the document]
Comment (Guismus) [reply cypress/commons/actions/brewery/BreweryParameters.js:185]: Done

File: cypress/commons/actions/generic/TableParameters.js (hunk 1 above)
Comment (Guismus) [reply cypress/commons/actions/generic/TableParameters.js:87]: DONE

File: cypress/commons/actions/generic/TableParameters.js (hunk 3 above)
Comment (Guismus) [reply cypress/commons/actions/generic/TableParameters.js:179]: We do not need it with new test

File: cypress/commons/actions/generic/TableParameters.js (hunk 2 above)
Comment (Guismus) [reply cypress/commons/actions/generic/TableParameters.js:153]: DONE

File: cypress/integration/brewery/TableParameters-fullscreen.spec.js (hunk 4 above)
Comment (Guismus) [reply cypress/integration/brewery/TableParameters-fullscreen.spec.js:51]: Yes, i fix it

File: cypress/integration/brewery/TableParameters-fullscreen.spec.js [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+    BreweryParameters.editCustomersTableStringCell('birthday', 2, '13/06/1988').should('have.text', '13/06/1988');
+
+    BreweryParameters.enterCustomersFullscreenButton();
+    TableParameters.getCell(getFullscreenTable(), 'age', 1, true).should('have.text', '9');
+    TableParameters.getCell(getFullscreenTable(), 'canDrinkAlcohol', 1).should('have.text', 'true');
+    TableParameters.getCell(getFullscreenTable(), 'favoriteDrink', 2).should('have.text', 'AppleJuice');
+    TableParameters.getCell(getFullscreenTable(), 'birthday', 2).should('have.text', '13/06/1988');
+    cy.get('body').type('{esc}');
+
+    BreweryParameters.importCustomersTableData('customers_thousand_line.csv');
```
[the remark this answers is not in the document]
Comment (Guismus) [reply cypress/integration/brewery/TableParameters-fullscreen.spec.js:77]: DONE

File: cypress/integration/brewery/TableParameters-fullscreen.spec.js (hunk 5 above)
Comment (Guismus) [reply cypress/integration/brewery/TableParameters-fullscreen.spec.js:44]: Now got the same, DONE

File: cypress/integration/brewery/TableParameters-fullscreen.spec.js (hunk 6 above)
Comment (Guismus) [reply cypress/integration/brewery/TableParameters-fullscreen.spec.js:62]: Default value of file (check if the modification in customer table do not affect customervip one)

File: cypress/fixtures/stubbing/TableParameters-fullscreen/solutions.js [hunk 10]
```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import { DEFAULT_SOLUTION } from '../default';
+
+const SOLUTION_WITH_TWO_TABLE = {
```
Comment (esasova) [inline cypress/fixtures/stubbing/TableParameters-fullscreen/solutions.js:6]: typo: table**S**

File: cypress/commons/constants/generic/IdConstants.js [hunk 11]
```diff
@@ -159,11 +159,16 @@ export const GENERIC_SELECTORS = {
...
+      toolbar: {
+        fullscreenButton: '[data-cy=grid-fullscreen-button]',
+      },
       header: '[class=ag-header-container]',
       colByName: '[col-id=$COLNAME]',
       rowsContainer: '[class=ag-center-cols-container]',
       row: '.ag-row',
       rowByIndex: '[row-index="$ROWINDEX"]',
+      fullscreenTable: '[data-cy=fullscreen-table]',
+      scroll: '[class=ag-body-vertical-scroll-viewport]',
```
Comment (esasova) [inline cypress/commons/constants/generic/IdConstants.js:171]: if I'm not mistaken, we don't need this constant anymore

File: cypress/fixtures/customers_thousand_line.csv
Comment (esasova) [inline cypress/fixtures/customers_thousand_line.csv:1]: this fixture file isn't used in test anymore

**Review by esasova — APPROVED** [review]
some comments but LGTM :+1:

File: cypress/fixtures/stubbing/TableParameters-fullscreen/solutions.js (hunk 10 above)
Comment (Guismus) [reply cypress/fixtures/stubbing/TableParameters-fullscreen/solutions.js:6]: DONE

File: cypress/commons/constants/generic/IdConstants.js (hunk 11 above)
Comment (Guismus) [reply cypress/commons/constants/generic/IdConstants.js:171]: DONE

**Review by csm-thu — APPROVED** [review]
LGTM :+1:

File: cypress/integration/brewery/TableParameters-fullscreen.spec.js [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+
+  beforeEach(() => {
+    Login.login();
+  });
+
+  after(() => {
+    stub.stop();
+  });
+
+  it('Edit a cell, and check if value as change in and out of fullscreen', () => {
```
[the remark this answers is not in the document]
Comment (esasova) [reply cypress/integration/brewery/TableParameters-fullscreen.spec.js]: yes, we talked about it with @Guismus, now, the test is shorter and no need to separate it :+1:

**Pull request merged**